webapps/frontend/app/viewmodels/wizards/addvdisk/gather.js

```diff
@@ -0,0 +1,131 @@
+        });
+
+        self.activate = function() {
+            generic.xhrAbort(self.loadStorageRoutersHandle);
+            self.loadStorageRoutersHandle = api.get('storagerouters', {
+                queryparams: {
+                    contents: 'storagedrivers',
+                    sort: 'name'
+                }
+            })
+                .done(function(data) {
+                    var guids = [], srdata = {};
+                    $.each(data.data, function(index, item) {
+                        guids.push(item.guid);
+                        srdata[item.guid] = item;
+                    });
+                    generic.crossFiller(
+                        guids, self.data.storageRouters,
+                        function(guid) {
+                            if (self.data.vPool() === undefined || !self.data.vPool().storageRouterGuids().contains(guid)) {
+                                return new StorageRouter(guid);
+                            }
+                        }, 'guid'
+                    );
+                    $.each(self.data.storageRouters(), function(index, storageRouter) {
+                        storageRouter.fillData(srdata[storageRouter.guid()]);
+                    });
+                    if (self.data.target() === undefined && self.data.storageRouters().length > 0) {
+                        self.data.target(self.data.storageRouters()[0]);
+                    }
+                });
+            if (generic.xhrCompleted(self.loadVPoolsHandle)) {
+                self.loadVPoolsHandle = api.get('vpools', {
+                        queryparams: {
+                        contents: ''
+                    }
+                })
+                    .done(function (data) {
+                        var guids = [], vpData = {};
+                        $.each(data.data, function (index, item) {
+                            guids.push(item.guid);
+                            vpData[item.guid] = item;
+                        });
+                        generic.crossFiller(
+                            guids, self.data.vPools,
+                            function (guid) {
+                                return new VPool(guid);
+                            }, 'guid'
+                        );
+                        $.each(self.data.vPools(), function (index, vpool) {
+                            if (guids.contains(vpool.guid())) {
+                                vpool.fillData(vpData[vpool.guid()]);
+                            }
+                        });
+                    });
+            }
+
+        }
+        self.finish = function() {
+            return $.Deferred(function(deferred) {
+                generic.alertInfo(
+                    $.t('ovs:wizards.add_vdisk.gather.started'),
+                    $.t('ovs:wizards.add_vdisk.gather.inprogress', { what: self.data.name() })
+                );
+                deferred.resolve();
```

Comment:
If you dont want to wait for the disk to be created, its still best to resolve afer launching the task